lib/chef/knife/bootstrap/chef-full.erb

```diff
@@ -19,32 +27,128 @@ exists() {
   fi
 }
 
+# do_wget URL FILENAME
+do_wget() {
+  echo "trying wget..."
+  wget -O "$2" "$1" 2>/tmp/stderr
```

Comment:
Here and in other places we likely want to use temporary file names more likely to be unique.  What are the portability requirements here? mktemp isn't part of POSIX but is supported on a number of systems.